lib/absinthe_auth/middleware.ex

```diff
@@ -29,22 +29,23 @@ defmodule AbsintheAuth.Middleware do
     Absinthe.Resolution.put_result(resolution, {:error, "Denied"})
   end
 
-  def maybe_push_middleware(%{state: :resolved} = resolution) do
+  def maybe_append_middleware(%{state: :resolved} = resolution) do
     # No need to the resolution middleware if we have already resolved
     resolution
   end
-  def maybe_push_middleware(%{definition: definition} = resolution) do
-    case resolution.middleware do
-      [_ | _] ->
-        # Additional middleware remain to be processed
-        # Let's continue
-        resolution
-
-      [] ->
-        field = definition.schema_node.identifier
-
-        # Insert default middleware
-        push_middleware(resolution, {Absinthe.Middleware.MapGet, field})
+  def maybe_append_middleware(%{definition: definition} = resolution) do
+    has_no_resolution_mware? = not Enum.any?(resolution.middleware, fn {mware_mod, _} ->
+      mware_mod == Absinthe.Middleware.MapGet ||
+      mware_mod == Absinthe.Resolution
+    end)
+    if Enum.empty?(resolution.middleware) && has_no_resolution_mware? do
```

Comment:
Do you need to check both of these? The former may be true but have no resolution middleware. So what if there is a non-resolving middleware set but no resolution middleware? The result would be no resolution set.